qiskit/algorithms/linear_solvers/matrices/tridiagonal.py

```diff
@@ -0,0 +1,210 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""Hamiltonian simulation of tridiagonal Toeplitz symmetric matrices."""
+
+from typing import Optional
+
+import numpy as np
+from scipy.sparse import diags
+
+from qiskit.circuit import QuantumCircuit, QuantumRegister, AncillaRegister, Qubit
+
+
+class Tridiagonal(QuantumCircuit):
+    """Class of tridiagonal Toeplitz symmetric matrices"""
+
+    def __init__(self, num_state_qubits: int, main_diag: float, off_diag: float,
+                 tolerance: Optional[float] = None, time: Optional[float] = None,
+                 trotter: Optional[int] = 1, name: str = 'tridi') -> None:
```

Comment:
I think `trotter: Optional[int] = 1` should just be `trotter: int = 1`. Having a default of 1 does not make it Optional. Optional is short for Union[None, x] where x is int in this case. I do not think None is a valid value.

Reply:
removed Optional from all values and set appropriate defaults